pkg/restapi/edv/operation/operations.go

```diff
@@ -236,11 +257,29 @@ func (c *Operation) registerHandler() {
 	c.handlers = []Handler{
 		support.NewHTTPHandler(createVaultEndpoint, http.MethodPost, c.createDataVaultHandler),
 		support.NewHTTPHandler(createDocumentEndpoint, http.MethodPost, c.createDocumentHandler),
-		support.NewHTTPHandler(retrieveDocumentEndpoint, http.MethodGet, c.retrieveDocumentHandler),
+		support.NewHTTPHandler(readDocumentEndpoint, http.MethodGet, c.readDocumentHandler),
 	}
 }
 
 // GetRESTHandlers get all controller API handler available for this service
 func (c *Operation) GetRESTHandlers() []Handler {
 	return c.handlers
 }
+
+// Unescapes the given path variable from the vars map and writes a response if any failure occurs.
+// Returns the unescaped version of the path variable and a bool indicating whether the unescaping was successful.
+func unescapePathVar(pathVar string, vars map[string]string, rw http.ResponseWriter) (string, bool) {
+	unescapedPathVar, err := url.PathUnescape(vars[pathVar])
+	if err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+
+		_, err = rw.Write([]byte(fmt.Sprintf("unable to escape %s path variable: %s", pathVar, err.Error())))
+		if err != nil {
+			log.Errorf("Failed to write response for %s unescaping failure: %s", pathVar, err.Error())
+		}
```

Comment:
should be in some common sendError(code, msg) function, I see this is repeated in many places.